src/west/app/project.py

```diff
@@ -640,6 +650,59 @@ def _die_if_manifest_project_filter(self, action):
                      'the manifest while projects are made inactive by the '
                      'project filter.')
 
+    def _untracked(self):
+        ppaths = []
+        untracked = []
+        for project in self._projects(None):
+            # We do not check for self.manifest.is_active(project) because
+            # inactive projects are still considered "tracked folders".
+            ppaths.append(Path(project.abspath).resolve(strict=False))
+
+        def _find_untracked(folder):
+            self.dbg(f'_find_untracked in: {folder}')
+            for e in [e.resolve() for e in folder.iterdir()]:
+                if e.is_dir():
+                    self.dbg(f'processing folder: {e}')
+                    for ppath in ppaths:
+                        # We cannot use samefile() because it requires the file
+                        # to exist (not always the case with inactive or even
+                        # uncloned projects).
+                        if ppath == e:
+                            # We hit a project root folder, skip it.
+                            break
+                        elif e in ppath.parents:
+                            self.dbg(f'recursing into: {e}')
+                            _find_untracked(e)
+                            break
+                    else:
+                        # No match, untracked folder.
+                        untracked.append(e)
+                        continue
+                else:
+                    # Untracked file.
+                    untracked.append(e)
+
+        # Since west supports nested projects (i.e. a project inside the folder
+        # of another project) we must sort the project paths to ensure that we
+        # hit the "enclosing" project first when iterating.
+        ppaths.sort()
```

Comment:
Please move this up, immediately after the first `ppaths` initialization step. It's confusing to have 25 lines between these two steps.